support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx

```diff
@@ -230,12 +231,49 @@ const isCardUserSelected = (
 		return false;
 	}
 	return (
-		(contributionType in paymentFrequencyMap &&
-			Number(urlSelectedAmount) === cardPrice) ||
+		Number(urlSelectedAmount) === cardPrice ||
 		Number(urlSelectedAmount) === cardPriceDiscount
 	);
 };
 
+function getCardData(
+	// TODO - this type could use some refinement
+	productDescription: (typeof productCatalogDescription)[keyof typeof productCatalogDescription],
+	pricing: number,
+	link: string,
+	promotion?: Promotion,
+	isRecommended = false,
+) {
+	return {
+		title: productDescription.label,
+		isRecommended,
+		isUserSelected: isCardUserSelected(pricing, promotion?.discount?.amount),
+		benefits: {
+			list: 'benefits' in productDescription ? productDescription.benefits : [],
+			description:
+				'benefitsSummary' in productDescription
+					? productDescription.benefitsSummary
+					: undefined,
+		},
+		planCost: {
+			price: pricing,
+			promoCode: promotion?.name,
+			discount:
+				promotion?.discount?.amount && promotion.discountedPrice
+					? {
+							percentage: promotion.discount.amount,
+							price: promotion.discountedPrice,
+							duration: {
+								value: promotion.numberOfDiscountedPeriods ?? 0,
```

Comment:
Should this read `value: promotion.discount.durationMonths ?? 0,` as no months coming through?